### PR TITLE
Minor enhancements to Scala Play server generator

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

- renamed to `scala-play-server` to make the naming more consistent with other generators
- included in the CircleCI integration tests
- add @adigerber as the author
- add copyright header
- add "beta" to the generator description

cc @clasnake (2017/07), @jimschubert (2017/09), @shijinkui (2018/01), @ramzimaalej (2018/03)

### CODE REVIEW

File: docs/generators/scala-play-framework.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
+---
+id: generator-opts-server-scala-play-framework
```
Comment (jimschubert) [inline docs/generators/scala-play-framework.md]: I think this file can be deleted.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaPlayFrameworkServerCodegen.java [hunk 2]
```diff
@@ -1,3 +1,19 @@
+/*
+ * Copyright 2018 OpenAPI-Generator Contributors (https://openapi-generator.tech)
```
Comment (jimschubert) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaPlayFrameworkServerCodegen.java]: Should the file name be updated to remove `Framework` as well?

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaPlayFrameworkServerCodegen.java]: I thought about that too but there are some other generators not following a similar naming convention. (looks like we never enforce it)

I may file another PR to clean those up before the 4.0.0 release.

File: docs/generators/scala-play-framework.md (hunk 1 above)
Comment (wing328) [reply docs/generators/scala-play-framework.md]: Done

**Pull request merged**